packages/backend/src/app/store-entry/store-entry.service.ts

```diff
@@ -6,27 +6,21 @@ const storeEntryRepo = databaseConnection.getRepository<StoreEntry>(StoreEntryEn
 
 export const storeEntryService = {
     async upsert({ projectId, request }: { projectId: ProjectId, request: PutStoreEntryRequest }): Promise<StoreEntry | null> {
-        const previousEntry = await this.getOne({ projectId, key: request.key })
-        if (previousEntry !== null) {
-            await storeEntryRepo.update(previousEntry.id, request)
-            return this.getOne({ projectId, key: request.key })
-        }
-        else {
-            const entryRequest: Omit<StoreEntry, 'created' | 'updated'> = {
-                id: apId(),
-                key: request.key,
-                value: request.value,
-                projectId,
-            }
-            return storeEntryRepo.save(entryRequest)
-        }
+        return storeEntryRepo.upsert({
+            id: apId(),
+            key: request.key,
+            value: request.value,
+            projectId,
+        }, ['projectId', 'key'])
```

Comment:
@abuaboud If the upsert doesn't return the same object, we need to use the lock. Because as I said before we can have a delete or another update in between.

Reply:
Hi @MarceloRGonc I think it's possible to construct the value from parameters (even better) construct it from InsertResut that returned from upsert

Reply:
Hi @abuaboud. The upsert doesn't work on this table. I'm getting this error: "there is no unique or exclusion constraint matching the ON CONFLICT specification". To use the upsert I think we need to change the table definition to make 'projectId' and 'Key' the primary key.

Reply:
I think we should add unique constraint on both projectId and key, Let's do that what do you think?

Reply:
@abuaboud and @khaledmashaly I already made the changes